fex2/collect.py

```diff
@@ -1,5 +1,77 @@
+from __future__ import annotations
+
 import csv
 import re
+from typing import Generator, Callable, Dict, List
+
+CONST_HEADER_KEYWORD = '[FEX2_HEADER] '
+CONST_EXPERIMENT_KEYWORD = '[FEX2_EXPERIMENT] '
+
+
+class Experiments:
+
+    def __init__(self, experiments_output: str, default_parsers=True):
+        self.__experiments_output = experiments_output
+        self.parsers: Dict[str, Callable[[Experiment], None]] = {}
+        if default_parsers:
+            self.parsers.update({
+                'benchmark': lambda experiment: get_run_argument('benchmark', experiment.header),
+                'thread_count': lambda experiment: get_run_argument('thread_count', experiment.header),
+                'type': lambda experiment: get_run_argument('type', experiment.header).split('_')[0],
+                'subtype': lambda experiment: "_".join(get_run_argument('type', experiment.header).split('_')[1:])})
```

Comment:
TBH, I'd rather not have a default parser here. The experiments could be very different and it's unclear what is a reasonable default.

Reply:
The default parser parses the fex header of each single experiment execution. Maybe rename the variable to default_header_parsers? :thinking:

Reply:
Ah, wait, it's the header. I misunderstood it. Yeah, in this case, it's totally fine.